docs/classification_algos.rst

```diff
@@ -11,21 +11,152 @@ Bayesian
+
+First, we create the base class `TreeNode class <https://github.com/bfortuner/ml-cheatsheet/blob/master/code/decision_tree.py#L7>`__ and  `DecisionTree <https://github.com/bfortuner/ml-cheatsheet/blob/master/code/decision_tree.py#L24>`__
+
+.. code-block:: python
+
+    class TreeNode:
+        def __init__(self, data_idx, depth, child_lst=[]):
+            self.data_idx = data_idx
+            self.depth = depth
+            self.child = child_lst
+            self.label = None
+            self.split_col = None
+            self.child_cate_order = None
+
+        def set_attribute(self, split_col, child_cate_order=None):
+            self.split_col = split_col
+            self.child_cate_order = child_cate_order
+
+        def set_label(self, label):
+            self.label = label
+..
+
+.. code-block:: python
+
+    class DecisionTree()
+        def fit(self, X, y):
+            """
+            X: train data, dimensition [num_sample, num_feature]
+            y: label, dimension [num_sample, ]
+            """
+            self.data = X
+            self.labels = y
+            num_sample, num_feature = X.shape
+            self.feature_num = num_feature
+            data_idx = list(range(num_sample))
+            # Set the root of the tree
+            self.root = TreeNode(data_idx=data_idx, depth=0, child_lst=[])
+            queue = [self.root]
+            while queue:
+                node = queue.pop(0)
+                # Check if the terminate criterion has been met
+                if node.depth>self.max_depth or len(node.data_idx)==1:
+                    # Set the label for the leaf node
+                    self.set_label(node)
+                else:
+                    # Split the node
+                    child_nodes = self.split_node(node)
+                    if not child_nodes:
+                        self.set_label(node)
+                    else:
+                        queue.extend(child_nodes)
+..
+
+For CART algorithm, when constructing the binary tree, it will try search for the feature and threshold that will yield the largest gain or the least impurity. The split criterion is a combination of the child nodes' impurity. For the child nodes' impurity, gini coefficient or information gain are adopted in classification. For regression problem, mean-square-error or mean-absolute-error are used. Example codes are showed below. For more details about the formulas, please refer to `Mathematical formulation for decision tree in scikit-learn documentation <https://scikit-learn.org/stable/modules/tree.html#mathematical-formulation>`__
```

Comment:
For CART algorithm, when constructing the binary tree, **it** will try search for
-->
**The** CART algorithm, when constructing the binary tree, will try search**ing** for